server/src/main/java/org/elasticsearch/cluster/metadata/MetadataMappingService.java

```diff
@@ -167,7 +167,7 @@ private static ClusterState applyRequest(
                 DocumentMapper mergedMapper = mapperService.merge(
                     MapperService.SINGLE_MAPPING_NAME,
                     mappingUpdateSource,
-                    MergeReason.MAPPING_UPDATE
+                    request.autoUpdate() ? MergeReason.MAPPING_AUTO_UPDATE : MergeReason.MAPPING_UPDATE
```

Comment:
should it not also use the same login in the call a few lines up [here](https://github.com/elastic/elasticsearch/blob/c0587289af31553af81e1dcfba2e7f959a27817e/server/src/main/java/org/elasticsearch/cluster/metadata/MetadataMappingService.java#L151)?

Reply:
Good catch. I'm not sure how https://github.com/elastic/elasticsearch/pull/96235 even worked without this 🤔

Reply:
Oh, I think that is because we're not validating the merged mapper in `MapperService#mergeMappings`, in contrast to `MapperService#merge`.